plugin/src/main/groovy/com/novoda/buildproperties/internal/BuildPropertiesException.groovy

```diff
@@ -32,14 +32,23 @@ class BuildPropertiesException extends Exception {
         List<Throwable> newExceptions = new ArrayList<>(exceptions)
         List<String> newAdditionalMessages = new ArrayList<>(additionalMessages)
         if (throwable instanceof BuildPropertiesException) {
-            newExceptions.addAll(((BuildPropertiesException) throwable).exceptions)
-            newAdditionalMessages.addAll(((BuildPropertiesException) throwable).additionalMessages)
+            def other = (BuildPropertiesException) throwable
+            newExceptions.addAll(removeDuplicateExceptions(other.exceptions))
+            newAdditionalMessages.addAll(removeDuplicateAdditionalMessages(other.additionalMessages))
         } else {
-            newExceptions.add(throwable)
+            newExceptions.addAll(removeDuplicateExceptions([throwable]))
         }
         return new BuildPropertiesException(newExceptions, newAdditionalMessages)
     }
 
+    private List<Throwable> removeDuplicateExceptions(List<Throwable> others) {
```

Comment:
I'd prefer passing in all parameters here (and below) and making the method static. Might be a matter of teatse, though.

Reply:
I was on the fence, I wrote it as static method then thought I'd try this instead. Will change 👍